src/library/FIFO.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * This is a First-In-First-Out (FIFO) library
```

Comment:
I would replace "library" with "class".

The "library" is a collection of utility JS classes.